frontend/src/views/NftDisplay.vue

```diff
@@ -245,6 +399,90 @@ export default Vue.extend({
       document.execCommand('copy');
       document.body.removeChild(dummy);
     },
+    convertWeiToSkill(wei: string): string {
+      return fromWeiEther(wei);
+    },
+    maxPrecisionSkill(listedPrice: string): string {
+      return this.convertStringToDecimal(
+        this.convertWeiToSkill(listedPrice),
+        8
+      );
+    },
+    convertStringToDecimal(val: string, maxDecimals: number) {
+      return new BigNumber(val).toFixed(maxDecimals);
+    },
+    async purchaseNft() {
+      this.marketOutcome = null;
+      if (this.nftId === null) return;
+
+      const price = await this.lookupNftPrice(this.nftId);
+      if (price) {
+        this.nftPrice = price;
+      } else {
+        return;
+      }
+
+      const skillChainPrice = this.convertStringToDecimal(
+        this.convertWeiToSkill(price),
+        2
+      );
+      const skillUIPrice = this.convertStringToDecimal(
+        this.nftPriceInSkill(),
+        2
+      );
+
+      if (skillChainPrice !== skillUIPrice) {
+        (this as any).$dialog.notify.error(
+          'The price of the listing has changed. Please refresh listing and try again'
+        );
+        return;
+      }
+
+      this.waitingMarketOutcome = true;
```

Comment:
This is not used at all. Either use it in template to show some kind of loading or remove.